### PR TITLE
CLOUDP-78867: [Milestone] Add option to get connection string(s)

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-78867

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

## Further comments

```
/bin/mongocli atlas cluster cs describe Cluster0 -o json
{
  "standard": "mongodb://cluster0-shard-00-00.ajlj3.mongodb-dev.net:27017,cluster0-shard-00-01.ajlj3.mongodb-dev.net:27017,cluster0-shard-00-02.ajlj3.mongodb-dev.net:27017/?ssl=true\u0026authSource=admin\u0026replicaSet=atlas-avx159-shard-0",
  "standardSrv": "mongodb+srv://cluster0.ajlj3.mongodb-dev.net"
}
```
```
./bin/mongocli atlas cluster cs get Cluster0
STANDARD CONNECTION STRING
mongodb+srv://cluster0.ajlj3.mongodb-dev.net
```

```
./bin/mongocli atlas cluster cs get Cluster0 --type private
PRIVATE CONNECTION STRING
mongodb+srv://cluster0.ajlj3.mongodb-dev.net

```

### CODE REVIEW

File: internal/cli/atlas/clusters/connetionstring/describe_test.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	}
+
+	mockStore.
+		EXPECT().
+		AtlasCluster(describeOpts.ProjectID, describeOpts.name).
+		Return(expected, nil).
+		Times(1)
+
+	err := describeOpts.Run()
+	if err != nil {
```
Comment (savannaendicott) [inline internal/cli/atlas/clusters/connetionstring/describe_test.go:48]: nitpick: should this format be used or `assert.NoError(t, err)` as you've implemented above?

File: e2e/atlas/clusters_test.go [hunk 2]
```diff
@@ -73,14 +67,8 @@ func TestClustersFlags(t *testing.T) {
...
 		resp, err := cmd.CombinedOutput()
-
-		if err != nil {
-			t.Fatalf("unexpected error: %v, resp: %v", err, string(resp))
-		}
-
-		if !strings.Contains(string(resp), "Cluster available") {
-			t.Errorf("got=%#v\nwant=%#v\n", string(resp), "Cluster available at:")
-		}
+		assert.NoError(t, err)
```
Comment (savannaendicott) [inline e2e/atlas/clusters_test.go:70]: Are we shifting from using `if err!= nil { t.Fatalf(...)}` to `assert.NoError`? If so, could you please provide information about this decision on the PR description? If not, could you explain the changes to this file?

Comment (andreaangiolillo) [reply e2e/atlas/clusters_test.go:70]: This shifting has already begun in the e2e tests (examples https://github.com/mongodb/mongocli/blob/master/e2e/atlas/alerts_test.go,
https://github.com/mongodb/mongocli/blob/master/e2e/atlas/custom_dns_test.go
https://github.com/mongodb/mongocli/blob/master/e2e/atlas/search_test.go). We generally try to use `assert` in the e2e tests where we have several `if` to make the code more readable. I suppose we could use it for unit tests as well but the impact would be less noticeable.

**Comment by cailinanne** [comment]
I don't have as strong opinion on this, but there seems to be little value in providing the  "STANDARD FORMAT" - all MongoDB drivers for MongoDB 3.6+ can understand the SRV format.

**Comment by andreaangiolillo** [comment]
@gssbzn ready for review

**Comment by andreaangiolillo** [comment]
[atlas_clusters_e2e](https://evergreen.mongodb.com/task/mongocli_master_e2e_atlas_clusters_atlas_clusters_e2e_patch_051076fad7c69c7488dbc78ae81ce5631148bd59_601007fb2a60ed5cc8be852a_21_01_26_12_15_56)

File: internal/cli/atlas/clusters/connectionstring/connection_string.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+package connectionstring
+
+import (
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/spf13/cobra"
+)
+
+func Builder() *cobra.Command {
+	const use = "connectionString"
```
Comment (gssbzn) [inline internal/cli/atlas/clusters/connectionstring/connection_string.go:23]: [nit] `GenerateAliases` needs the usage in plural
```suggestion
	const use = "connectionStrings"
```

File: e2e/atlas/clusters_test.go [hunk 4]
```diff
@@ -73,14 +68,8 @@ func TestClustersFlags(t *testing.T) {
...
-
-		if err != nil {
-			t.Fatalf("unexpected error: %v, resp: %v", err, string(resp))
-		}
-
-		if !strings.Contains(string(resp), "Cluster available") {
-			t.Errorf("got=%#v\nwant=%#v\n", string(resp), "Cluster available at:")
-		}
+		a.NoError(err)
+		a.Contains(string(resp), "Cluster available", fmt.Sprintf("got=%#v\nwant=%#v\n", string(resp), "Cluster available at:"))
```
Comment (gssbzn) [inline e2e/atlas/clusters_test.go:72]: [nit] no need to provide the error message
```suggestion
		a.Contains(string(resp), "Cluster available")
```

File: e2e/atlas/clusters_test.go [hunk 5]
```diff
@@ -20,22 +20,20 @@ import (
...
-	}
-	clusterName, e := RandClusterName()
-	if e != nil {
-		t.Fatalf("unexpected error: %v", e)
-	}
+	cliPath, err := e2e.Bin()
+	a := assert.New(t)
+	a.NoError(err)
+	clusterName, err := RandClusterName()
+	a.NoError(err)
```
Comment (gssbzn) [inline e2e/atlas/clusters_test.go:35]: I'm ok if you want to keep migrating tests to use assert but want thing you need to be aware is that this change is not really equivalent and may introduce weird tests on error if you see the original test had a `t.Fatalf` which means if there was an error the rest of the test would not run, this is important as getting some errors should prevent further testing given they will fail either way.

With your change now if there's an error on line 33, line 34 is still being check, please consider making sure that if we have irrecuperable errors we actually finish the test early, there are examples around our e2e tests using `ifs` to do this

File: internal/cli/atlas/clusters/connectionstring/description.go [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package connectionstring
+
+const (
+	connectionString = "Manage MongoDB cluster connection string"
+	describe         = "Retrieve the connection string of your MongoDB cluster."
```
Comment (gssbzn) [inline internal/cli/atlas/clusters/connectionstring/description.go:18]: let's make this clear 
```suggestion
	describe         = "Retrieve the SRV connection string of your MongoDB cluster."
```

File: internal/cli/atlas/clusters/connectionstring/describe.go [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+func DescribeBuilder() *cobra.Command {
+	opts := &DescribeOpts{}
+	cmd := &cobra.Command{
+		Use:     "describe <clusterName>",
+		Aliases: []string{"get"},
+		Short:   describe,
+		Args:    require.ExactArgs(1),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			opts.OutWriter = cmd.OutOrStdout()
+			opts.Template = describeTemplateStandard
```
Comment (gssbzn) [inline internal/cli/atlas/clusters/connectionstring/describe.go:67]: [q] `opts.InitOutput`?

File: internal/usage/usage.go [hunk 8]
```diff
@@ -39,6 +39,7 @@ const (
 	Forever                         = "Acknowledge an alert “forever”."
 	Status                          = "Alert's status."
 	Until                           = "Acknowledged until a date."
+	ConnectionStringType            = "When set to 'private' retrieves the private connection string. "
```
Comment (gssbzn) [inline internal/usage/usage.go:42]: [nit] let's be more specific here 
```suggestion
	ConnectionStringType            = "When set to 'private' retrieves the Network-peering-endpoint-aware connection string. "
```

**Review by gssbzn — CHANGES_REQUESTED** [review]
A couple of copy changes, also the e2e migration to use assert need some work to make the migration equivalent, please be mindful where we used to do fatal would not run the next steps, please keep this behaviour. Also I see in some of the assertions you provide the error message, you can mostly avoid this given what you're providing is almost the same as the default behaviour of assert

File: e2e/atlas/clusters_test.go [hunk 9]
```diff
@@ -20,22 +20,22 @@ import (
...
-		t.Fatalf("unexpected error: %v", e)
-	}
-	clusterName, e := RandClusterName()
-	if e != nil {
-		t.Fatalf("unexpected error: %v", e)
-	}
+	cliPath, err := e2e.Bin()
+	a := assert.New(t)
+	req := require.New(t)
+	req.NoError(err)
```
Comment (gssbzn) [inline e2e/atlas/clusters_test.go:35]: 👍

**Review by gssbzn — APPROVED** [review]
LGTM thanks for the test fixes

**Pull request merged**